File: app/src/main/java/org/deiverbum/app/model/MassReadingList.java
package org.deiverbum.app.model;

import static org.deiverbum.app.utils.Constants.TITLE_MASS_READING;
import static org.deiverbum.app.utils.Utils.LS2;

import android.text.SpannableStringBuilder;

import org.deiverbum.app.utils.Utils;

import java.util.Collections;
import java.util.List;

public class MassReadingList extends Liturgy {
    public List<Evangelio> evangelios;

    public List<Evangelio> getEvangelio() {
        return evangelios;
    }

    List<MassReading> lecturas;

    public List<MassReading> getLecturas() {
        return lecturas;
    }

    public void setLecturas(List<MassReading> lecturas) {
        this.lecturas = lecturas;
    }

    public void setEvangelio(List<Evangelio> evangelios) {
        this.evangelios = evangelios;
    }

    public MetaLiturgia getMetaLiturgia() {
        return metaliturgia;
    }

    public void setMetaLiturgia(MetaLiturgia metaliturgia) {
        this.metaliturgia = metaliturgia;
    }

    private SpannableStringBuilder getTitulo() {
        return Utils.toH3Red(Utils.toUpper(TITLE_MASS_READING));
    }

    public String getTituloForRead() {
        return Utils.pointAtEnd(TITLE_MASS_READING);
    }

    @SuppressWarnings("unused")
    private MetaLiturgia metaliturgia;

    public SpannableStringBuilder getAllForView() {
        SpannableStringBuilder sb = new SpannableStringBuilder("");
        for (MassReading b : lecturas) {
            sb.append(b.getAll());
        }
        return sb;
    }

    @SuppressWarnings("unused")
    public SpannableStringBuilder getAllEvangelioForView() {
        SpannableStringBuilder sb = new SpannableStringBuilder("");
        for (MassReading b : lecturas) {
            if (b.getOrden() >= 40) {
                sb.append(b.getAll());
            }
        }
        return sb;
    }

    public SpannableStringBuilder getAllEvangelioForRead() {
        SpannableStringBuilder sb = new SpannableStringBuilder("");
        for (MassReading b : lecturas) {
            if (b.getOrden() >= 40) {
                sb.append(b.getAllForRead());
            }
        }
        return sb;
    }

    public void sort() {
        Collections.sort(this.lecturas);
    }

    public SpannableStringBuilder getForView() {
        SpannableStringBuilder sb = new SpannableStringBuilder("");
        try {
            //sb.append(hoy.getForViewMisa());
            sb.append(hoy.getAllForView());
            sb.append(LS2);
            sb.append(getTitulo());
            sb.append(LS2);
            for (MassReading l : lecturas) {
                sb.append(l.getAll());
            }
        } catch (Exception e) {
            sb.append(Utils.createErrorMessage(e.getMessage()));
        }
        return sb;
    }

    @Override
    public SpannableStringBuilder getAllForRead() {

        SpannableStringBuilder sb = new SpannableStringBuilder("");
        try {
            sb.append(hoy.getAllForRead());
            sb.append(getTituloForRead());
            for (MassReading l : lecturas) {
                sb.append(l.getAllForRead());
            }
        } catch (Exception e) {
            sb.append(Utils.createErrorMessage(e.getMessage()));
        }
        return sb;
    }
}

File: app/src/main/java/org/deiverbum/app/model/MassReadingList.kt
package org.deiverbum.app.model

import android.text.SpannableStringBuilder
import org.deiverbum.app.domain.model.TodayRequest
import org.deiverbum.app.utils.ColorUtils
import org.deiverbum.app.utils.Constants
import org.deiverbum.app.utils.Utils

class MassReadingList : Liturgy() {
    @JvmField
    var type = 0
    var lecturas: List<MassReading?>? = null
    private val titulo: SpannableStringBuilder
        get() = Utils.toH3Red(Utils.toUpper(Constants.TITLE_MASS_READING))
    override val tituloForRead: String
        get() = Utils.pointAtEnd(Constants.TITLE_MASS_READING)
    val allForView: SpannableStringBuilder
        get() {
            val sb = SpannableStringBuilder("")
            for (b in lecturas!!) {
                sb.append(b?.getAll())
            }
            return sb
        }

    fun sort() {
        //Collections.sort(lecturas)
    }

    fun getForView(todayRequest:TodayRequest): SpannableStringBuilder {
        ColorUtils.isNightMode = todayRequest.isNightMode

        val sb = SpannableStringBuilder("")
        try {
            //sb.append(hoy.getForViewMisa());
            //sb.append(today?.singleForView)
            sb.append(Utils.LS2)
            sb.append(titulo)
            sb.append(Utils.LS2)
            for (l in lecturas!!) {
                sb.append(l?.getAll(type))
            }
        } catch (e: Exception) {
            sb.append(Utils.createErrorMessage(e.message))
        }
        return sb
    }

    //@Override
    val allForRead: StringBuilder
        get() {
            val sb = StringBuilder()
            try {
                sb.append(today?.getSingleForRead())
                sb.append(tituloForRead)
                for (l in lecturas!!) {
                    sb.append(l?.getAllForRead(type))
                }
            } catch (e: Exception) {
                sb.append(Utils.createErrorMessage(e.message))
            }
            return sb
        }
}